discovery-provider/src/wsgi.py

```diff
@@ -1,9 +1,23 @@
 # exposes flask create_app() function as gunicorn entrypoint
 import logging
+from os import getenv
 
 from src.app import create_app
 
 logger = logging.getLogger(__name__)
 
+if not getenv("PROMETHEUS_MULTIPROC_DIR"):
+    logger.error("PROMETHEUS_MULTIPROC_DIR is not set")
```

Comment:
is the idea here that third party nodes shouldn't startup if this config isn't set?

Reply:
Correct! It should be set now within `discovery-provider/Dockerfile`, but figured to be extra safe.

Reply:
i think this should be optional tbqh - our nodes can have it but it seems tangential to the core of running a node @dmanjunath

Reply:
We need the envar set for #1: https://github.com/prometheus/client_python#multiprocess-mode-eg-gunicorn

I figured to include this if/exit just in case we don't include `$PROMETHEUS_MULTIPROC_DIR` in our k8s manifests, or something, so that we can detect the issue early instead of allowing us to run with crippled metrics in the future.

Perhaps there's a better way to break early?

Reply:
i think the best analogy for the metrics should be like log sharing - optional , maybe encouraged - but not a barrier to running a node entirely

Reply:
Sure thing, I'm +0 on making `/metrics` optional, in which case we can have an if-switch around this line: https://github.com/AudiusProject/audius-protocol/pull/2504/files#diff-02d33986bb6b5cc6dd8c27ac4d95dbb2b2a9a402bbbdfcbd0b1f0a1eed326a7fR334

However, `$PROMETHEUS_MULTIPROC_DIR` is a required variable if we choose to enable log sharing while using gunicorn.

What are your thoughts on how to proceed with both conditions?

Reply:
ah i see, i do agree w/u let's not if/switch on that. definitely can always set the variable regardless of the pod actually running with each deployment. will resolve this thread